dns_cache.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"github.com/rs/dnscache"
+)
+
+// Local DNS cache because in this world things are ephemeral
+var dnsCache = &dnscache.Resolver{}
+
+// How often should we check for successful updates to cached entries
+const dnsCacheRefreshInterval = 5 * time.Minute
+
+func init() {
+	// Configure DNS cache to not remove stale records to protect gateway from
+	// catastrophic failures like https://github.com/ipfs/bifrost-gateway/issues/34
+	options := dnscache.ResolverRefreshOptions{}
+	options.ClearUnused = false
+	options.PersistOnFailure = false
```

Comment:
i think this should be `PersistOnFailure = true` in order to not remove stale records to protect from failures

Reply:
```suggestion
	options.PersistOnFailure = true
```